doc/source/user_guide/getting_help.rst

```diff
@@ -28,6 +28,25 @@ The ``quick search`` field located in the navigation bar of the html
 documentation can be used to search for specific keywords (segmentation,
 rescaling, denoising, etc.).
 
+API Discovery
+-------------
+
+NumPy provides a ``lookfor`` function to search API functions. 
+By default ``lookfor`` will search the NumPy API.
+NumPy lookfor example:
+```np.lookfor('eigenvector') ```
+
+But it can be used to search in modules, by passing in the module
```

Comment:
Let's give the users one-and-one-way-only of doing this.

I've also left a comment on the original bug report: is it worth adding `skimage.lookfor`?